### PR TITLE
switch for consistently enabling gradient tracking

### LINKED ISSUE #2661: add config arg for inplace scatter
### Describe the task. It can be a feature, documentation, etc.

for xai work:

We are using WeatherGenerator for Layer-wise Relevance Propagation (LRP) interpretability work. This requires computing gradients with respect to the ERA5 input tokens, rather than the model weights, so the autograd computation graph needs to remain intact through the forward pass.
While looking at this workflow, we noticed a small autograd-related point in EmbeddingEngine.forward in engines.py that is relevant for interpretability use cases. The current code is:
tokens_all.scatter_(0, scatter_idxs, torch.cat(x_embeds) + pe_embed[pe_idxs])
 
Our understanding is that, because tokens_all is a requires_grad=False tensor, the in-place scatter_ operation does not preserve the autograd graph back to the ERA5 input tokens. The forward-pass values appear to be unchanged; the difference is in whether gradients can be tracked for interpretability workflows.
We have tested a few possible ways of handling this locally, and would welcome your guidance on which approach would best fit the WeatherGenerator codebase.

Agreed upon config flag change

### Hedgedoc URL, if you are keeping notes, plots, logs in hedgedoc.

_No response_

### Area

- [ ] datasets, data readers, data preparation and transfer
- [x] model
- [x] science
- [ ] infrastructure and engineering
- [ ] evaluation, export and visualization
- [ ] documentation
- [ ] performance

### PR DESCRIPTION
**Pull request by enssow**
## Description

In the current WeatherGenerator setup when the model does a forward pass, it runs through `_scatter` in some instances this does not preserve the autograd graph back to the ERA5 input tokens. I've implemented a switch in the config. Will add some sample runs to show that performance of model doesn't change here.

## Issue Number

Closes #2661 

Is this PR a draft? Mark it as draft.

## Checklist before asking for review

-   [ ] I have performed a self-review of my code
-   [ ] My changes comply with basic sanity checks:
      - I have fixed formatting issues with `./scripts/actions.sh lint`
      - I have run unit tests with `./scripts/actions.sh unit-test`
      - I have documented my code and I have updated the docstrings.
      - I have added unit tests, if relevant
-   [ ] I have tried my changes with data and code:
      - I have run the integration tests with `./scripts/actions.sh integration-test`
      - (bigger changes) I have run a full training and I have written in the comment the run_id(s): `launch-slurm.py --time 60`
      - (bigger changes and experiments) I have shared a hegdedoc in the github issue with all the configurations and runs for this experiments
-   [ ] I have informed and aligned with people impacted by my change:
    - for config changes: the MatterMost channels and/or a design doc
    - for changes of dependencies: the MatterMost software development channel

### CODE REVIEW

File: config/default_config.yml [hunk 1]
```diff
@@ -92,6 +92,8 @@ load_chkpt: {}
 norm_type: "LayerNorm"
 qk_norm_type: null  # if null, defaults to norm_type
 
+lrp_enabled: False
```
Comment (clessig) [inline config/default_config.yml:95]: The meaning of config options should be self-explanatory, at least for someone familiar with the model. This is not the case here.

File: src/weathergen/model/engines.py [hunk 2]
```diff
@@ -123,8 +123,12 @@ def forward(self, batch, pe_embed):
             scatter_idxs = self.get_scatter_idxs_vectorized(batch)
             scatter_idxs = scatter_idxs.unsqueeze(1).repeat((1, self.cf.ae_local_dim_embed))
 
-            # actual scatter operation and apply per cell positional encoding
-            tokens_all.scatter_(0, scatter_idxs, torch.cat(x_embeds))
+            # for xai lrp technique 
+            if self.cf.lrp_enabled:
+                tokens_all = tokens_all.scatter(0, scatter_idxs, torch.cat(x_embeds))
+            else:
+                # actual scatter operation and apply per cell positional encoding
```
Comment (clessig) [inline src/weathergen/model/engines.py:130]: The comment should be above the if-statement, the comment in l126 above l128

**Review by clessig — COMMENTED** [review]
Thanks for the contribution. Minor changes required.
